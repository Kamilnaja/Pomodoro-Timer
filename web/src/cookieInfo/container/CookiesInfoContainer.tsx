import React, { Component } from 'react';
import { connect } from 'react-redux';
import { SettingsState } from '../../settings/store/interfaces/settingsInterfaces';
import { CookiesInfoComponent } from '../component/CookiesInfoComponent';
import { CookiesInfoContainerProps } from './CookiesInfoContainerProps';
import { handleSaveSettings, handleGetSettings } from 'settings/store/actions/settingsActions';
import { AuthState } from '../../auth/store/interfaces/authState';
import { Loader } from '../../shared/loader/Loader';
import { ErrorComponent } from '../../shared/error/errorComponent/ErrorComponent';

class CookiesInfoContainer extends Component<CookiesInfoContainerProps> {
  componentDidMount() {
    if (this.props.authState.isLoggedIn) {
      this.props.handleGetSettings();
    }
  }

  handleAcceptCookieConsent = () => {
    this.props.handleSaveSettings({ ...this.props.settingsState.settings, isCookieConsentAccepted: true });
  };

  render() {
    const { settingsState } = this.props;

    if (settingsState.isLoading) {
      return <Loader />;
    } else if (settingsState.error) {
      return <ErrorComponent />;
    } else if (!settingsState.settings.isCookieConsentAccepted) {
      return (
        <CookiesInfoComponent
          settingsState={this.props.settingsState}
          handleSaveCookieConsent={this.handleAcceptCookieConsent}
        />
      );
    } else {
      return <></>; // todo - check if needed
    }
  }
}

const mapStateToProps = (state: { settings: SettingsState; auth: AuthState }) => {
  const settingsState = state.settings;
  const authState = state.auth;
  return { settingsState, authState };
};

const mapDispatchToProps = {
  handleSaveSettings,
  handleGetSettings,
};

export default connect(mapStateToProps, mapDispatchToProps)(CookiesInfoContainer);
