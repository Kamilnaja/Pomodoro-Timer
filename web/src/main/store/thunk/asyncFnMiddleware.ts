export async function fetchStats(dispatch: any, getState: any) {
  const response = await client.get("/fakeApi/todos");
  dispatch({ type: "todos/todosLoaded", payload: response.todos });
}
