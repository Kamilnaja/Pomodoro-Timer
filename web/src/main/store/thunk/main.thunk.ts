import TodayStatistics from "../../../shared/store/interfaces/todayStatistics.interface";
import {
  getTodayStatistics,
  getTodayStatisticsError,
  getTodayStatisticsSuccess,
  savePomodoro,
  savePomodoroError,
  savePomodoroSuccess,
} from "../actions/actions";
// todo - different versions for prod etc
const API_URL = "http://localhost:8080/pomodoros";

function makePostRequest() {
  return fetch(API_URL, {
    method: "POST",
  });
}

export function makeGetRequest() {
  return fetch(API_URL);
}

function handleErrors(response: any): Promise<any> {
  if (!response.ok) {
    throw Error(response.statusText);
  }
  return response.json();
}

export function savePomodoroThunk() {
  return (dispatch: Function) => {
    dispatch(savePomodoro());

    return makePostRequest()
      .then(handleErrors)
      .then(() => dispatch(savePomodoroSuccess()))
      .catch((error) => dispatch(savePomodoroError(error)));
  };
}

export function getPomodorosThunk() {
  return (dispatch: Function) => {
    dispatch(getTodayStatistics());

    return makeGetRequest()
      .then(handleErrors)
      .then((payload: TodayStatistics) => {
        console.log(payload);
        return dispatch(getTodayStatisticsSuccess(payload));
      })
      .catch((error) => dispatch(getTodayStatisticsError(error)));
  };
}
