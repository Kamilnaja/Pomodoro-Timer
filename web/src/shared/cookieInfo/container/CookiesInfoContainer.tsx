import React, { Component } from 'react';
import { connect } from 'react-redux';
import { handleGetSettings, handleSaveSettings, hideCookieInfo } from 'settings/store/actions/settingsActions';
import { AuthState } from '../../../auth/store/interfaces/authState';
import { SettingsState } from '../../../settings/store/interfaces/settingsInterfaces';
import { isCookieConsentVisible } from '../../../settings/store/selectors/settingsSelectors';
import { ErrorComponent } from '../../error/errorComponent/ErrorComponent';
import { Loader } from '../../loader/Loader';
import { isCookieConsentAcceptedKey } from '../../settings/initialConfig';
import { CookiesInfoComponent } from '../component/CookiesInfoComponent';
import { CookiesInfoContainerProps } from './CookiesInfoContainerProps';

class CookiesInfoContainer extends Component<CookiesInfoContainerProps> {
  componentDidMount() {
    if (this.props.authState.isLoggedIn) {
      this.props.handleGetSettings();
    } else {
      if (localStorage.getItem(isCookieConsentAcceptedKey)) {
        this.props.hideCookieInfo();
      }
    }
  }

  handleAcceptCookieConsent = () => {
    if (!this.props.authState.isLoggedIn) {
      window.localStorage.setItem(isCookieConsentAcceptedKey, JSON.stringify(true));
      this.props.hideCookieInfo();
    } else {
      this.props.handleSaveSettings({ ...this.props.settingsState.settings, isCookieConsentAccepted: true });
    }
  };

  render() {
    const { settingsState } = this.props;
    if (settingsState.error) {
      return <ErrorComponent />;
    } else if (isCookieConsentVisible(settingsState)) {
      return (
        <CookiesInfoComponent
          settingsState={this.props.settingsState}
          handleSaveCookieConsent={this.handleAcceptCookieConsent}
        />
      );
    } else {
      return null;
    }
  }
}

const mapStateToProps = (state: { settings: SettingsState; auth: AuthState }) => {
  const settingsState = state.settings;
  const authState = state.auth;
  return { settingsState, authState };
};

const mapDispatchToProps = {
  handleSaveSettings,
  handleGetSettings,
  hideCookieInfo,
};

export default connect(mapStateToProps, mapDispatchToProps)(CookiesInfoContainer);
