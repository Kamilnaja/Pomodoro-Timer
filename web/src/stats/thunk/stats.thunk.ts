import { handleErrors } from "../../main/store/thunk/main.thunk";
import { initialConfig } from "../../shared/settings/initialConfig";
import TodayStatistics from "../../shared/store/interfaces/todayStatistics.interface";
import {
  getTodayStatistics,
  getTodayStatisticsError,
  getTodayStatisticsSuccess,
} from "../store/actions/stats.actions";

const API_URL = initialConfig.apiUrl;

const makeGetRequest = () => fetch(API_URL + "/pomodoros_done_today");

export const getPomodorosThunk = () => (dispatch: Function) => {
  dispatch(getTodayStatistics());

  return makeGetRequest()
    .then(handleErrors)
    .then((response) => response.json())
    .then((payload: TodayStatistics) => {
      return dispatch(getTodayStatisticsSuccess(payload));
    })
    .catch((error) => dispatch(getTodayStatisticsError(error)));
};
