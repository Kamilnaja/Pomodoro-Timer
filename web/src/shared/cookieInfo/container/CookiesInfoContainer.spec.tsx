import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import CookiesInfoContainer from './CookiesInfoContainer';

describe('CookiesInfoContainer', () => {
  let store;
  let component: renderer.ReactTestRenderer;
  const mockStore = configureStore([thunk]);

  beforeEach(() => {
    store = mockStore({
      settings: {
        error: '',
      },
      auth: {
        isLoggedIn: false,
      },
    });

    component = renderer.create(
      <Provider store={store}>
        <CookiesInfoContainer />
      </Provider>,
    );
  });

  it('should match snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
