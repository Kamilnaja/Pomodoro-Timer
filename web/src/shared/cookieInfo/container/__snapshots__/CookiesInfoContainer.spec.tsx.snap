// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookiesInfoContainer should match snapshot 1`] = `null`;
